File: 3layerJava/Shop3layer/src/ProductBUS.java
import java.sql.ResultSet;
import java.util.ArrayList;

public class ProductBUS {
    public static ArrayList<Product> getAll() {
        ArrayList<Product> products = new ArrayList<>();
        try {
            ResultSet rs = ProductDAO.getAll();
            while (true) {
                assert rs != null;
                if (!rs.next()) break;
                Product product = new Product();
                product.setId(rs.getInt(1));
                product.setThumbnail(rs.getString(2));
                product.setName(rs.getString(3));
                product.setDescription(rs.getString(4));
                product.setPrice(rs.getFloat(5));
                products.add(product);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return products;
    }

    public static int add(Product product) {
        return ProductDAO.add(product);
    }

    public static int update(int id, Product product) {
        return ProductDAO.update(id, product);
    }

    public static int delete(int id) {
        return ProductDAO.delete(id);
    }
}

File: 3layer/frontend/assets/js/ProductBUS.js
const { ProductDAO } = require('./js/ProductDAO.js');

class ProductBUS {
    constructor() {
        this.productDAO = new ProductDAO();
    }

    async getAll() {
        return await this.productDAO.getAllProducts();
    }

    async getById(id) {
        return await this.productDAO.getProductById(id);
    }

    async create(product) {
        return await this.productDAO.addProduct(product);
    }

    async update(id, product) {
        return await this.productDAO.updateProduct(id, product);
    }

    async delete(id) {
        return await this.productDAO.deleteProduct(id);
    }
}

module.exports = { ProductBUS };